docs/_source/practical_guides/annotation_workflows/semantic_search.md

```diff
@@ -0,0 +1,361 @@
+
+Over the years, many approaches have been used to turn text into numerical representations. The goal is to "encode" meaning, context,  topics, etc.. This can be used to find "semantically" similar text. Some of these approaches are *LSA* (Latent Semantic Analysis), *tf-idf*, *LDA* (Latent Dirichlet Allocation), or *doc2Vec*. More recent methods fall in the category of "neural" methods, which leverage the power of large neural networks to *embed* text into dense vectors (a large array of real numbers). These methods have demonstrated a great ability to capture semantic features. These methods are powering a new wave of technologies that fall under categories like neural search, semantic search, or vector search. Most of these methods involve using a large language model to encode the full context of a textual snippet, such as a sentence, a paragraph, and more lately larger documents.
+
+```{note}
+
+In the context of Argilla, we intentionally use the term `vector` in favor of `embedding` to emphasize that users can leverage methods other than neural, which might be cheaper to compute or be more useful for their use cases.
+```
+
+In the next sections, we show how to encode text using different models and services and how to add them to Argilla records.
+
+```{warning}
+
+If you run into issues when logging records with large vectors using `rg.log`, we recommend you to use a smaller `chunk_size` as shown in the following examples.
+```
+
+### Sentence Transformers
+
+SentenceTransformers is a Python framework for state-of-the-art sentence, text and image embeddings. There are dozens of [pre-trained models available](https://huggingface.co/models?pipeline_tag=sentence-similarity&sort=downloads) on the Hugging Face Hub.
+
+The code below will load a dataset from the Hub, encode the `text` field, and create the `vectors` field which will contain only one key (`mini-lm-sentence-transformers`).
+
+```{note}
+
+Vector keys are arbitrary names that will be used as a name for the vector and shown in the UI if there's more than 1 so users can decide which vector to use for finding similar records. Remember you can associate several vectors to one record by using different keys.
+```
+
+```{warning}
+Due to the vector dimension limitation of Elasticsearch and Opensearch Lucene-based engines, currently, you cannot register vectors with dimensions greater than `1024`.
+```
+
+To run the code below you need to install `sentence_transformers` and `datasets` with pip: `pip install sentence_transformers datasets`
+
+```python
+from sentence_transformers import SentenceTransformer
+
+from datasets import load_dataset
+
+# Define fast version of sentence transformers
+encoder = SentenceTransformer("all-MiniLM-L6-v2", device="cpu")
```

Comment:
you can maybe use the tailorAI smaller embedding model thingy we mentioned in the machine learning channel?